src/modules/navigator/mission_block.cpp

```diff
@@ -1004,12 +1030,16 @@ void MissionBlock::startPrecLand(uint16_t land_precision)
 {
 	if (_mission_item.land_precision == 1) {
 		_navigator->get_precland()->set_mode(PrecLandMode::Opportunistic);
-		_navigator->get_precland()->on_activation();
+		send_vehicle_command(vehicle_command_s::VEHICLE_CMD_DO_SET_MODE, 1, PX4_CUSTOM_MAIN_MODE_AUTO,
```

Comment:
The assumption is that you're at the end of the mission or RTL so you can just exit the mode.
It also tells the user in QGC that they are now in precision land so that's a small bonus.

> It also tells the user in QGC that they are now in precision land so that's a bonus :D 

yea, it's the quick way.

I think the src id and compid would still be the fmu and dmesg would say from internal command instead of external. I can double check. So we can still know it didn't come from a control station or from any external mavlink command.

Reply:
However, it would show up in the ulog the same way a commander precland shell command.